allensdk/brain_observatory/behavior/behavior_project_cache/project_apis/data_io/behavior_ecephys_project_cloud_api.py

```diff
@@ -0,0 +1,336 @@
+
+        self.logger = logging.getLogger("BehaviorEcephysProjectCloudApi")
+        self._get_ecephys_session_table()
+        self._get_behavior_session_table()
+        self._get_unit_table()
+        self._get_probe_table()
+        self._get_channel_table()
+
+    @classmethod
+    def from_s3_cache(cls, cache_dir: Union[str, Path],
+                      bucket_name: str,
+                      project_name: str,
+                      ui_class_name: str) -> "BehaviorEcephysProjectCloudApi":
+        """instantiates this object with a connection to an s3 bucket and/or
+        a local cache related to that bucket.
+
+        Parameters
+        ----------
+        cache_dir: str or pathlib.Path
+            Path to the directory where data will be stored on the local system
+
+        bucket_name: str
+            for example, if bucket URI is 's3://mybucket' this value should be
+            'mybucket'
+
+        project_name: str
+            the name of the project this cache is supposed to access. This
+            project name is the first part of the prefix of the release data
+            objects. I.e. s3://<bucket_name>/<project_name>/<object tree>
+
+        ui_class_name: str
+            Name of user interface class (used to populate error messages)
+
+        Returns
+        -------
+        BehaviorProjectCloudApi instance
+
+        """
+        cache = S3CloudCache(cache_dir,
+                             bucket_name,
+                             project_name,
+                             ui_class_name=ui_class_name)
+        return cls(cache)
+
+    @staticmethod
+    def from_local_cache(
+        cache_dir: Union[str, Path],
+        project_name: str,
+        ui_class_name: str,
+        use_static_cache: bool = False
+    ) -> "BehaviorEcephysProjectCloudApi":
+        """instantiates this object with a local cache.
+
+        Parameters
+        ----------
+        cache_dir: str or pathlib.Path
+            Path to the directory where data will be stored on the local system
+
+        project_name: str
+            the name of the project this cache is supposed to access. This
+            project name is the first part of the prefix of the release data
+            objects. I.e. s3://<bucket_name>/<project_name>/<object tree>
+
+        ui_class_name: str
+            Name of user interface class (used to populate error messages)
+
+        Returns
+        -------
+        BehaviorProjectCloudApi instance
+
+        """
+        if use_static_cache:
+            cache = StaticLocalCache(
+                cache_dir,
+                project_name,
+                ui_class_name=ui_class_name
+            )
+        else:
+            cache = LocalCache(
+                cache_dir,
+                project_name,
+                ui_class_name=ui_class_name
+            )
+        return BehaviorEcephysProjectCloudApi(cache, local=True)
+
+    def get_behavior_session(
```

Comment:
Since we are not yet providing NWB files for behavior-only data, this method needs to fail with some helpful warning message when there is no `ecephys_session_id` that matches the specified `behavior_session_id` (there are going to be a lot more rows in `behavior_sessions.csv` than there are in `ecephys_sessions.csv`).

Reply:
done